platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java

```diff
@@ -368,6 +378,12 @@ enum OverlappingOutputHandling {
      */
     default void validate(WorkValidationContext validationContext) {}
 
+    default void checkDependencies(WorkValidationContext validationContext) {}
+
+    default FileCollectionStructureVisitor getInputFileChecker(WorkValidationContext validationContext) {
+        return FileCollectionStructureVisitor.NO_OP;
+    }
```

Comment:
💭 Also could find a better name for this perhaps. This is supposed to run the dependency check on file inputs as they are visited for snapshotting.

Reply:
I've improved the naming here, too.